libdevice/fallback-cstring.cpp

```diff
@@ -47,8 +47,8 @@ void *__devicelib_memcpy(void *dest, const void *src, size_t n) {
   if (dest == NULL || src == NULL || n == 0)
     return dest;
 
-  unsigned long dest_addr = reinterpret_cast<unsigned long>(dest);
-  unsigned long src_addr = reinterpret_cast<unsigned long>(src);
+  unsigned long long dest_addr = reinterpret_cast<unsigned long long>(dest);
```

Comment:
I think the right way to fix the warning is to use `std::uintptr_t` here.
```suggestion
  std::uintptr_t dest_addr = reinterpret_cast<std::uintptr_t>(dest);
```

Reply:
Done.